java-frontend/src/main/java/org/sonar/java/UCFGJavaVisitor.java

```diff
@@ -335,19 +337,23 @@ private LocationInFile location(SyntaxToken firstToken, SyntaxToken lastToken) {
     );
   }
 
-  private static boolean isStringVarDecl(Tree tree) {
+  private static boolean isObjectVarDeclaration(Tree tree) {
     if (!tree.is(Tree.Kind.VARIABLE)) {
       return false;
     }
 
     VariableTree var = (VariableTree) tree;
-    return isString(var.type().symbolType());
+    return isObject(var.type().symbolType());
   }
 
   private static boolean isString(Type type) {
     return type.is("java.lang.String");
   }
 
+  private static boolean isObject(Type type) {
+    return type.isSubtypeOf("java.lang.Object");
```

Comment:
probably better in this case to use the negation of the "isPrimitive" method : more efficient and closer to the definition of what we are trying to achieve.